libtsuba/src/RDGPartHeader.cpp

```diff
@@ -316,12 +338,54 @@ katana::from_json(const json& j, katana::RDGPartHeader& header) {
   j.at(kPartPropertyFilesKey).get_to(header.part_prop_info_list_);
   j.at(kPartPropertyMetaKey).get_to(header.metadata_);
 
+  /// Storage Format Version Handling
+
+  // support loading "storage_format_version=1" RDGs, aka RDGs without
+  // an explicit storage_format_version
   if (auto it = j.find(kStorageFormatVersionKey); it != j.end()) {
     it->get_to(header.storage_format_version_);
   } else {
     header.storage_format_version_ = kPartitionStorageFormatVersion1;
   }
 
+  // load the unstable_storage_format flag if it is present in the RDG
+  // RDGs created before support for unstable_storage_format was added do not have this flag.
+  // If it is not present, the RDG is assumed to *NOT* be stored in an unstable_storage_format since the rdg was created before unstable formats were introduced
+  if (auto it = j.find(kUnstableStorageFormatFlagKey); it != j.end()) {
+    it->get_to(header.unstable_storage_format_);
+    // Ensure unstable_storage_format RDGs are not loaded when the feature flag is not set
```

Comment:
So this should return an error?

Reply:
Apologies, that comment should have been on the if-else block below. Fixed.